hollow/src/main/java/com/netflix/hollow/api/codegen/delegate/HollowObjectDelegateCachedImplGenerator.java

```diff
@@ -173,62 +173,132 @@ public String generate() {
         return builder.toString();
     }
 
-    private void addAccessor(StringBuilder builder, FieldType fieldType, String fieldName) {
+    private void addAccessor(StringBuilder builder, FieldType fieldType, String fieldName, boolean reference) {
         switch(fieldType) {
         case BOOLEAN:
-            builder.append("    public boolean get").append(uppercase(fieldName)).append("(int ordinal) {\n");
-            builder.append("        if(").append(fieldName).append(" == null)\n");
-            builder.append("            return false;\n");
-            builder.append("        return ").append(fieldName).append(".booleanValue();\n");
-            builder.append("    }\n\n");
-            builder.append("    public Boolean get").append(uppercase(fieldName)).append("Boxed(int ordinal) {\n");
-            builder.append("        return ").append(fieldName).append(";\n");
-            builder.append("    }\n\n");
+            if(restrictApiToFieldType) {
+                if(!reference) {
+                    builder.append("    public boolean get").append(uppercase(fieldName)).append("(int ordinal) {\n");
```

Comment:
Perhaps we can improve this a bit without duplicating builder.append lines. For example:

if (!restrictApiToFieldType || !reference) {
   ... primitive
}

if (!restrictApiToFieldType || reference) {
   ... boxed
}

Reply:
This is hopefully a stop gap till V3, when we default to this flag, (and/or separate out the client-api generation).